applications/ecallmgr/src/ecallmgr_fs_xml.erl

```diff
@@ -404,6 +406,13 @@ route_resp_ringback(JObj) ->
             action_el(<<"set">>, <<"ringback=", (kz_term:to_binary(Stream))/binary>>)
     end.
 
+-spec maybe_emmit_aleg_callid() -> kz_types:xml_els().
+maybe_emmit_aleg_callid() ->
+    case ?ALEG_CALLID_HEADER of
+        'undefined' -> [];
+        HeaderName -> [action_el(<<"set">>, <<"sip_h_", HeaderName/binary, "=${uuid}">>)]
```

Comment:
@sergey-safarov Isn't freeswitch's UUID parameter the call-id of that leg?  Not sure how ${uuid} on the leg issuing the fetch request (usually the aleg) will translate to the aleg. Futher, custom SIP headers should start with "X-", iirc sip_h_* does not automatically do that.  Finally, this should be optional and default to disabled because it will expand the size of the UDP packet and fragmentation can be a real issue for some networks if it exceeds the MTU.

My untested recommendation would be something like this:

maybe_emmit_other_leg_callid() ->
    case  kapps_config:get_is_true(?APP_NAME, <<"emmit_other_leg_callid">>, 'false') of
        'false' -> [];
        'true' -> [action_el(<<"set">>, <<"sip_h_X-Other-Leg-Call-ID=${Other-Leg-Unique-ID}">>)]
    end.

However, I am not sure if the var replacement will be dynamic or only when the sip_h_* var is set.  Might need some tweaking but perhaps this helps show what I am thinking :)

Let me know!

Reply:
@k-anderson here `ALEG_CALLID_HEADER` is not param in fetch request/responce.
`ALEG_CALLID_HEADER` ecallmgr configuration param.
```erlang
-define(ALEG_CALLID_HEADER, kapps_config:get_ne_binary(?APP_NAME, <<"aleg_callid_header">>)).
```
This value is semi permanent for all calls and not affect fetch request/response.

`ALEG_CALLID_HEADER` is optional already. Because header default value is `undefined` (feature is disabled). In this case FreeSwitch dialplan not generate additional XML elements.

Var replacement is dynamic. Works well.

P.S. Commit about 'X-' headers will be added.

Reply:
Also about UDP size.
All X-headers will be striped when call is routed to phone via kamailio.
[Kamailio config link](https://github.com/2600hz/kazoo-configs-kamailio/blob/55091d53807d6d9034c692499eafb20e957077e3/kamailio/default.cfg#L682)

For carrier this not matter because they not use NAT or have properly configured NAT equpment.